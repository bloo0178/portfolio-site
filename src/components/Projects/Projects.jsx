import React from "react";
import styles from "./Projects.module.scss";
import Card from "../Card/Card";
import axios from 'axios';

// eslint-disable-next-line
let sampleData = [
  {
    title: "Title",
    description: `Quo alii voluptua apeirian in. Ex modus noster nostrum sed. Qui no
  paulo platonem, has te ridens praesent.`,
    source: "#",
    demo: "#"
  }
];

class Projects extends React.Component {
  state = {
    projects: []
  }

  componentDidMount() {
    axios.get(`https://7c0gcbb2ua.execute-api.us-east-1.amazonaws.com/prod`)
    .then(res => {
      console.log(res.data);
      const projects = res.data.Items; 
      this.setState({ projects })
    })
  }

  projectCards = () => (
    <>
    {this.state.projects.map(project => (
      <Card 
      key={project.projectId}
      description={project.description}
      title={project.name}
      demoUrl={project.demoUrl}
      srcUrl={project.codeUrl}
      />
    ))}
    </>
  )

  render() {
    return (
      <div className={styles.container}>
        <div>
          <h2>Projects</h2>
        </div>
        <div className={styles.cards}>
        {this.projectCards()}
          {/*<Card
            title="test"
            description="Quo alii voluptua apeirian in. Ex modus noster nostrum sed. Qui no
            paulo platonem, has te ridens praesent."
            demoURL="jeffbloom.io"
            srcURL="jeffbloom.io"
          />*/}
        </div>
      </div>
    );
  }
}

export default Projects;
