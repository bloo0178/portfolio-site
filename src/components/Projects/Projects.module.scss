@import "../../common/_variables";

.container {
  background-color: transparent;
  //min-height: 100vh;
  padding-top: 4rem;
  padding-bottom: 4rem;
  padding-left: 4rem;
  padding-right: 4rem;
  display: flex; 
  flex-direction: column;
}

.cards {
  display: flex;
  justify-content: center;
  align-content: center;
  flex-wrap: wrap;
}
