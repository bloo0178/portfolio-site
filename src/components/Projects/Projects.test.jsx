import React from "react";
import {shallow} from "enzyme";
import Projects from "./Projects";

const state = [
	{
		codeUrl: "https://github.com/bloo0178/sendbird-chat-client",
		demoUrl: "https://sendbird-chat.jeffbloom.io/",
		name: "SendBird Chat App",
		projectId: "001",
		tech: "React, AWS (Amplify), SendBird Chat SDK, Material UI"
	}
];

describe("Projects component", () => {
	test("it matches the snapshot", () => {
        let wrapper = shallow(<Projects />);
        wrapper.setState(state);
		expect(wrapper).toMatchSnapshot();
	});
});
