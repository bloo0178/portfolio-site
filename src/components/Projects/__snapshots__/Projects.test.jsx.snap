// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Projects component it matches the snapshot 1`] = `
<div
  className="container"
>
  <h1>
    Projects
  </h1>
  <div
    className="cards"
  />
</div>
`;
